src/wasm-interpreter.h

```diff
@@ -2758,7 +2759,17 @@ class ModuleRunnerBase : public ExpressionRunner<SubType> {
     auto* func = wasm.getFunction(curr->target);
     Flow ret;
     if (func->imported()) {
-      ret.values = externalInterface->callImport(func, arguments);
+      // The call.without.effects intrinsic is a call to an import that actually
+      // calls the given function reference that is the final argument.
+      if (func->module == Intrinsics::BinaryenIntrinsicsModule &&
+          func->base == Intrinsics::CallWithoutEffects) {
```

Comment:
It looks like we could use `Intrinsics::isCallWithoutEffects` for this check, and we could do it even before checking `if (func->imported())`. That would let us leave the `func->imported()` case unmodified.

Reply:
Sounds good, done.